sdk/python/lib/test/provider/experimental/test_analyzer.py

```diff
@@ -117,6 +118,47 @@ def __init__(self, args: Args): ...
     )
 
 
+def test_analyze_component_complex_type():
+    class ComplexType:
+        value: pulumi.Input[str]
+        optional_value: Optional[pulumi.Input[int]]
```

Comment:
Do we also want to test doc->description translations? (applies to other tests too)

Reply:
Definitely, this reminds me that there is some extra work I need to port from the POC to this code, for handling doc strings for attributes.

Tracked here https://github.com/pulumi/pulumi/issues/18438